### PR TITLE
adding nightly feature for no_std use of ereport function from libstd

### PR DESCRIPTION
**Pull request by Gautamgith**
Support for no_std use of ereport from libstd as nightly feature.

### CODE REVIEW

File: sgx-isa/Cargo.toml [hunk 1]
```diff
@@ -22,3 +22,4 @@ bitflags = "1" # MIT/Apache-2.0
 [features]
 large_array_derive = []
 sgxstd = []
+nightly = []
```
Comment (parthsane) [inline sgx-isa/Cargo.toml]: Add a newline here

File: sgx-isa/src/lib.rs [hunk 2]
```diff
@@ -694,3 +700,83 @@ fn test_eq() {
...
+            : /* no output registers */
+            : "{eax}"(ENCLU_EREPORT),
+              "{rbx}"(targetinfo),
+              "{rcx}"(reportdata),
+              "{rdx}"(report.as_mut_ptr())
+        );
+
+        report.assume_init()
+    }
+}
```
Comment (parthsane) [inline sgx-isa/src/lib.rs]: Add a newline here

File: sgx-isa/src/lib.rs [hunk 3]
```diff
@@ -694,3 +700,83 @@ fn test_eq() {
     b.miscmask = 0xdeadbeef;
     assert!(a == b);
 }
+
+
+
+
+//##################################################################################################################################################################################################################################
```
Comment (parthsane) [inline sgx-isa/src/lib.rs]: Remove unnecessary comment and newlines

**Comment by parthsane** [comment]
please rebase from master.

**Comment by parthsane** [comment]
Please rebase from master

File: sgx-isa/src/lib.rs [hunk 4]
```diff
@@ -694,3 +697,77 @@ fn test_eq() {
...
+#[unstable(feature = "sgx_platform", issue = "56975")]
+pub struct Align512<T>(pub T);
+
+const ENCLU_EREPORT: u32 = 0;
+const ENCLU_EGETKEY: u32 = 1;
+
+/// Call the `EGETKEY` instruction to obtain a 128-bit secret key.
+#[cfg(all(feature = "nightly", target_env = "sgx"))]
+#[unstable(feature = "sgx_platform", issue = "56975")]
+pub fn egetkey(request: &Align512<[u8; 512]>) -> Result<Align16<[u8; 16]>, u32> {
```
Comment (jethrogb) [inline sgx-isa/src/lib.rs]: These shouldn't be exposed as free functions, instead, these should be exposed the same way the `std::os::fortanix_sgx` functions are used now (see elsewhere in this file)

File: sgx-isa/src/lib.rs [hunk 5]
```diff
@@ -10,6 +10,9 @@
 //! convenient.
 //!
 //! [isdm]: https://www-ssl.intel.com/content/www/us/en/processors/architectures-software-developer-manuals.html
+#![feature(asm)]
+#![feature(lang_items)]
+#![feature(no_std)]
```
Comment (jethrogb) [inline sgx-isa/src/lib.rs]: These should all be dependent on the nightly feature being set. Also, we shouldn't need `lang_items` and `no_std`.

File: sgx-isa/src/lib.rs [hunk 6]
```diff
@@ -20,12 +23,12 @@
 #[cfg(all(feature = "sgxstd", target_env = "sgx"))]
 extern crate std;
 
+#[cfg(all(feature = "nightly", target_env = "sgx"))]
+extern crate core;
```
Comment (jethrogb) [inline sgx-isa/src/lib.rs]: This should come by default since `no_std` is set

File: sgx-isa/src/lib.rs (hunk 4 above)
Comment (parthsane) [reply sgx-isa/src/lib.rs]: Should it be in a different file in the same project in rust-sgx repo, or should it be in the rustlang repo?

Comment (jethrogb) [reply sgx-isa/src/lib.rs]: It should be in this file, or perhaps in a submodule.

**Comment by jethrogb** [comment]
Closing in favor of #185

**Pull request closed without merging**